inform7/Internal/Inter/Architecture32Kit/Sections/Input Output.i6t

```diff
@@ -196,16 +191,12 @@ to document all of that.
     return res;
 ];
 
-[ VM_KeyDelay tenths  key done ix;
+[ VM_KeyDelay tenths  key done;
     glk_request_char_event(gg_mainwin);
     glk_request_timer_events(tenths*100);
     while (~~done) {
         glk_select(gg_event);
-        ix = HandleGlkEvent(gg_event, 1, gg_arguments);
-        if (ix == 2) {
-            key = gg_arguments-->0;
-            done = true;
-        } else if (ix >= 0 && gg_event-->0 == 1 or 2) {
+        if (gg_event-->0 == evtype_Timer or evtype_CharInput) {
```

Comment:
This is the sort of usage of `glk_select` that I was talking about as problematic -- it will just drop any other event type on the floor without letting it be handled.

*Anywhere* that `glk_select` is called, it needs to admit the possibility that some other event might be returned, then be able to go off and do some other processing, and then either resume or cancel based on the result of such other processing (and possibly `glk_cancel_*` its local requests before going off to that external processing, and resume them after).

Reply:
@uecasm Missed this comment sorry. The glk event handling process is now *inside* the `glk_select` function, so that before this line runs the entire process has already been completed. Glk handling rules are able to convert one type of glk event to another, or convert it to a null event in order to prevent the callers of `glk_select` from seeing the event.
See https://github.com/ganelson/inform/blob/94f13ce59ec8736802c65cb1046979500853c399/inform7/Internal/Inter/Architecture32Kit/Sections/Glk.i6t#L251

Reply:
Ah, I see, that does appear to be an improvement.

It's unfortunate about the `debug_rules` though; sometimes you really do want to debug why your event handlers might be misbehaving...

Reply:
Hmm. I guess it would be possible to only conditionally disable debug handling if text input requests are still active. But I feel like that would probably be more confusing for authors. Instead, if authors do need `debug_rules` to be on in the glk event handling rules, then they can probably just turn it on themselves in a "first" rule.